user-management/src/main/java/de/tum/in/www1/artemis/usermanagement/web/rest/AccountResource.java

```diff
@@ -0,0 +1,272 @@
+    }
+
+    /**
+     * {@code GET  /authenticate} : check if the user is authenticated, and return its login.
+     *
+     * @param request the HTTP request.
+     * @return the login if the user is authenticated.
+     */
+    @GetMapping("/authenticate")
+    public String isAuthenticated(HttpServletRequest request) {
+        log.debug("REST request to check if the current user is authenticated");
+        return request.getRemoteUser();
+    }
+
+    /**
+     * {@code GET  /account} : get the current user.
+     *
+     * @return the current user.
+     * @throws EntityNotFoundException {@code 404 (User not found)} if the user couldn't be returned.
+     */
+    @GetMapping("/account")
+    public UserDTO getAccount() {
+        long start = System.currentTimeMillis();
+        User user = userRepository.getUserWithGroupsAuthoritiesAndGuidedTourSettings();
+        UserDTO userDTO = new UserDTO(user);
+        log.info("GET /account {} took {}ms", user.getLogin(), System.currentTimeMillis() - start);
+        return userDTO;
+    }
+
+    /**
+     * GET /account/password : get the current users password.
+     *
+     * @return the ResponseEntity with status 200 (OK) and the current user password in body, or status 500 (Internal Server Error) if the user couldn't be returned
+     */
+    @GetMapping(value = "/account/password", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> getPassword() {
+        // This method is used to show the password for users that have been generated automatically based on LTI
+        // It only allows to decrypt and return the password of internal users and only of the currently logged in user
+        Map<String, String> body = new HashMap<>();
+        body.put("password", passwordService.decryptPasswordOfCurrentUser());
+        return new ResponseEntity<>(body, HttpStatus.OK);
+    }
+
+    /**
+     * {@code PUT  /account} : update the current user information.
+     *
+     * @param userDTO the current user information.
+     * @throws EmailAlreadyUsedException {@code 400 (Bad Request)} if the email is already used.
+     * @throws RuntimeException {@code 500 (Internal Server Error)} if the user login wasn't found.
+     */
+    @PutMapping("/account")
+    public void saveAccount(@Valid @RequestBody UserDTO userDTO) {
+        if (isRegistrationDisabled()) {
+            throw new AccessForbiddenException("User Registration is disabled");
+        }
+        final String userLogin = SecurityUtils.getCurrentUserLogin().orElseThrow(() -> new InternalServerErrorException("Current user login not found"));
+        Optional<User> existingUser = userRepository.findOneByEmailIgnoreCase(userDTO.getEmail());
+        if (existingUser.isPresent() && (!existingUser.get().getLogin().equalsIgnoreCase(userLogin))) {
+            throw new EmailAlreadyUsedException();
+        }
+        Optional<User> user = userRepository.findOneByLogin(userLogin);
+        if (user.isEmpty()) {
```

Comment:
this one also can be handled with orElseThrow, right? you just did the same thing at line 197